pyiron/lammps/control.py

```diff
@@ -201,42 +201,57 @@ def pressure_to_lammps(self, pressure, rotation_matrix):
         Lammps handles complex cells in a particular way, namely by using an upper triangular cell. This means we may
         need to convert our pressure tensor to a new coordinate frame. We also handle that transformation here.
 
+        I case of a single pressure value, it is again returned as a single pressure value, to be used with the "iso"
+        option (i.e., coupled deformation in x, y, and z).
+
         Finally, we also ensure that the units are converted from pyiron's GPa to whatever Lammps needs.
 
         Args:
             pressure (float/list/tuple/numpy.ndarray): The pressure(s) to convert.
             rotation_matrix (numpy.ndarray): The 3x3 matrix rotating from the pyiron to Lammps coordinate frame.
 
         Returns:
-            (numpy.ndarray): pxx, pyy, pzz, pxy, pxz, and pyz to be passed to Lammps.
+            (list): pxx, pyy, pzz, pxy, pxz, and pyz to be passed to Lammps.
+
+            or
+
+            (float): a single, isotropic pressure to be used with the "iso" option
         """
 
-        # Get something with six elements
-        if not hasattr(pressure, "__len__"):
-            if not np.isclose(np.matrix.trace(rotation_matrix), 3) and pressure == 0.0:
-                pressure = np.append(pressure * np.ones(3), pressure * np.ones(3))
-            else:
-                pressure = np.append(pressure * np.ones(3), 3 * [None])
-        else:
-            if len(pressure) > 6:
-                raise ValueError("Pressure can have a maximum of 6 values, (x, y, z, xy, xz, and yz), but got " +
-                                 "{}".format(len(pressure)))
-            all_pressures = 6 * [None]
-            all_pressures[:len(pressure)] = pressure
-            pressure = np.array(all_pressures, dtype=float)
-
-        # Cell degrees of freedom can be kept fixed using None for the pressure, check where that's done.
-        not_none_mask = (pressure!=None)
-        if not np.any(not_none_mask):
+        # If pressure is a scalar, only unit conversion is needed.
+        if np.isscalar(pressure):
+            return float(pressure) * LAMMPS_UNIT_CONVERSIONS[self["units"]]["pressure"]
+
+        # Normalize pressure to a list of 6 entries.
+        if len(pressure) > 6:
+            raise ValueError("Pressure can have a maximum of 6 values, (x, y, z, xy, xz, and yz), but got " +
```

Comment:
Is not this part a bit tricky? Let's say I want to transfer the external pressure from one simulation to the next, at the end of the simulation I get the pressure as 3x3 and now I manually have to transform it to 6x1? Can we included the 3x3 to 6x1 conversion here?

Reply:
That behavior is not new and it is the documented API (I just fiddled a bit with the implementation). As documented in the docstring, the API never allowed 6x6. Since this PR fixes a bug, could such discussion be moved to a separate issue, since this one is accumulating feature creep?

I actually agree in theory, but don't really care enough since the conversion to Voigt is trivial. I would prefer that someone who cares does the work, which includes tests, documentation, and likely a bunch of discussion. Meanwhile, my simulations don't work.